src/main/2.0/scala/com/holdenkarau/spark/testing/receiver/InputStreamTestingContext.scala

```diff
@@ -0,0 +1,65 @@
+package com.holdenkarau.spark.testing.receiver
+
+import org.apache.spark.SparkContext
+import org.apache.spark.streaming.dstream.DStream
+import org.apache.spark.streaming.{Duration, Seconds, StreamingContext}
+
+/**
+  * Sequences the execution of user-provided code that comprises one individual test of an
+  * [[InputStreamSuiteBase]] test suite. Each individual test will typically instantiate one instance of this class.
+  *
+  * @param sparkContext        - a  [[com.holdenkarau.spark.testing.SharedSparkContext]] instance, provided by
+  *                            [[com.holdenkarau.spark.testing.StreamingSuiteBase]], which will be used to create
+  *                            an instance of  a [[org.apache.spark.streaming.StreamingContext]] that will be used
+  *                            for a run of one individual test.
+  *
+  * @param dStreamCreationFunc - a user-provided code block that uses  the StreamingContext provided by this class
+  *                            to generate a [[org.apache.spark.streaming.dstream.DStream]].
+  *
+  * @param testDataGenerationFunc - a user-provided code block that generates test data to be consumed by the
+  *                               [[org.apache.spark.streaming.dstream.InputDStream]] under test.
+  *
+  * @param pauseDuration          - the amount of time to wait before kicking off [[testDataGenerationFunc]].
+  *
+  * @param expectedResult         - the number of results that we expect to find in the DStream produced
```

Comment:
So this is a list, I think a better description might be "A list containing the expected results on the output DStream" or something similar, since right now it sounds like this would be an int field or something.